lib/Dialect/AIEX/Transforms/AIEDmaToNpu.cpp

```diff
@@ -588,7 +559,8 @@ struct WriteBdToBlockWritePattern : OpConversionPattern<NpuWriteBdOp> {
       words[3] |= op.getD0Stride() & 0xfffff;
 
       // DMA_BDX_4
-      words[4] = burstLengthEncoding; // burst length;
+      words[4] = (getShimBurstLengthEncoding(tm, op.getBurstLength()) & 0x3)
+                 << 30;
```

Comment:
I've moved the left shift out of the "encoding" returned by the target model into this code because this is where all the other mask and shift information is encoded in code.